### PR TITLE
Fix inconsistency in kubernetes commands / arguments handling between jvm and native

### LINKED ISSUE #24885: `quarkus.openshift.arguments` fails on OCP + native mode
### Describe the bug

When I deploy a native application that requires custom java arguments into OCP 4.10 the container crash at creation/start time with the following error:

```
oc get pods 
app-1-dwfb5    0/1     CreateContainerError   0          20s 
```

```
oc describe pod app-1-dwfb5 

Events:
  Type     Reason          Age               From               Message
  ----     ------          ----              ----               -------
  Normal   Scheduled       63s               default-scheduler  Successfully assigned ts-fgrltckoih/app-1-dwfb5 to ocp4-10-gl2nw-worker-0-nbgvv
  Normal   AddedInterface  61s               multus             Add eth0 [10.128.2.216/23] from openshift-sdn
  Normal   Pulled          59s               kubelet            Successfully pulled image "image-registry.openshift-image-registry.svc:5000/ts-fgrltckoih/app@sha256:525c030e520217f4e049496e4a4a9d26afa93efae750fe36f7c2f38348254efc" in 2.076644547s
  Warning  Failed          58s               kubelet            Error: container create failed: time="2022-04-12T07:51:38Z" level=error msg="runc create failed: unable to start container process: exec: \"ARG1\": executable file not found in $PATH"
  Normal   Pulled          57s               kubelet            Successfully pulled image "image-registry.openshift-image-registry.svc:5000/ts-fgrltckoih/app@sha256:525c030e520217f4e049496e4a4a9d26afa93efae750fe36f7c2f38348254efc" in 139.839247ms
  Warning  Failed          57s               kubelet            Error: container create failed: time="2022-04-12T07:51:39Z" level=error msg="runc create failed: unable to start container process: exec: \"ARG1\": executable file not found in $PATH"
  Warning  Failed          43s               kubelet            Error: container create failed: time="2022-04-12T07:51:53Z" level=error msg="runc create failed: unable to start container process: exec: \"ARG1\": executable file not found in $PATH"
  Normal   Pulled          43s               kubelet            Successfully pulled image "image-registry.openshift-image-registry.svc:5000/ts-fgrltckoih/app@sha256:525c030e520217f4e049496e4a4a9d26afa93efae750fe36f7c2f38348254efc" in 129.635937ms
  Normal   Pulled          31s               kubelet            Successfully pulled image "image-registry.openshift-image-registry.svc:5000/ts-fgrltckoih/app@sha256:525c030e520217f4e049496e4a4a9d26afa93efae750fe36f7c2f38348254efc" in 138.761777ms
  Warning  Failed          31s               kubelet            Error: container create failed: time="2022-04-12T07:52:05Z" level=error msg="runc create failed: unable to start container process: exec: \"ARG1\": executable file not found in $PATH"
  Normal   Pulled          20s               kubelet            Successfully pulled image "image-registry.openshift-image-registry.svc:5000/ts-fgrltckoih/app@sha256:525c030e520217f4e049496e4a4a9d26afa93efae750fe36f7c2f38348254efc" in 229.794292ms
  Warning  Failed          20s               kubelet            Error: container create failed: time="2022-04-12T07:52:16Z" level=error msg="runc create failed: unable to start container process: exec: \"ARG1\": executable file not found in $PATH"
  Normal   Pulling         9s (x6 over 61s)  kubelet            Pulling image "image-registry.openshift-image-registry.svc:5000/ts-fgrltckoih/app@sha256:525c030e520217f4e049496e4a4a9d26afa93efae750fe36f7c2f38348254efc"
  Normal   Pulled          9s                kubelet            Successfully pulled image "image-registry.openshift-image-registry.svc:5000/ts-fgrltckoih/app@sha256:525c030e520217f4e049496e4a4a9d26afa93efae750fe36f7c2f38348254efc" in 128.577373ms
  Warning  Failed          9s                kubelet            Error: container create failed: time="2022-04-12T07:52:27Z" level=error msg="runc create failed: unable to start container process: exec: \"ARG1\": executable file not found in $PATH"
```

Appliction.properties:

```
quarkus.openshift.arguments=ARG1,ARG2 
```

Main class:
```
@QuarkusMain
public final class Main {
    private static final Logger LOG = Logger.getLogger(Main.class);
    private static final String ARGUMENTS_FROM_MAIN = "Received arguments: ";

    private Main() {

    }

    public static void main(String... args) {
        LOG.info(ARGUMENTS_FROM_MAIN + String.join(",", args));
        Quarkus.run(args);
    }
}
```

Endpoint

```
@Path("/args")
public class ArgsResource {
    @Inject
    @CommandLineArguments
    String[] args;

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String get() {
        return Stream.of(args).collect(Collectors.joining(","));
    }
} 
```

Works in JVM mode fails in native mode (over OCP)

### How to reproduce

Requirements:

* Java 11/ 17
* Quarkus 2.7.5.Final / 2.7.5.ER2
* Openshift 4.10

Reproducer:

```
git clone git@github.com:quarkus-qe/quarkus-test-suite.git 
```

Module: lifecycle-application

```
mvn -B -V clean verify -Dnative -Dopenshift -fae -Dall-modules -pl lifecycle-application 
```

### Expected behavior

Same as JVM mode

### PR DESCRIPTION
**Pull request by iocanel**
The pull request addresses the inconsitencies between jvm and native mode when it comes to command / arguments handling.

With this change the s2i builder image is no longer taken into consideration.
The command may still be `generated` if the user provides arguments without command.

Resolves: #24885

### CODE REVIEW

File: extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java [hunk 1]
```diff
@@ -132,6 +132,9 @@ public void openshiftRequirementsJvm(OpenshiftConfig openshiftConfig,
         String baseJvmImage = config.baseJvmImage
                 .orElse(OpenshiftConfig.getDefaultJvmImage(compiledJavaVersion.getJavaVersion()));
 
+        boolean hasCustomJarPath = config.jarFileName.isPresent() || config.jarDirectory.isPresent();
+        boolean hasCustomJvmArugments = config.jvmArguments.isPresent();
```
Comment (gsmet) [inline extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java:136]:
```suggestion
        boolean hasCustomJvmArguments = config.jvmArguments.isPresent();
```

File: extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java [hunk 2]
```diff
@@ -180,6 +184,9 @@ public void openshiftRequirementsNative(OpenshiftConfig openshiftConfig,
 
         String nativeBinaryFileName = null;
 
+        boolean hasCustomNativePath = config.nativeBinaryFileName.isPresent() || config.nativeBinaryDirectory.isPresent();
+        boolean hasCustomNativeArugments = config.nativeArguments.isPresent();
```
Comment (gsmet) [inline extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java:188]:
```suggestion
        boolean hasCustomNativeArguments = config.nativeArguments.isPresent();
```

**Review by gsmet — DISMISSED** [review]
Spotted some typos in variable names. Use your IDE to fix them as I only suggested the first occurrence.

File: extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java [hunk 3]
```diff
@@ -150,14 +153,15 @@ public void openshiftRequirementsJvm(OpenshiftConfig openshiftConfig,
...
             //In all other cases its the responsibility of the image to set those up correctly.
-            if (!baseImage.isPresent()) {
+            if (hasCustomJarPath || hasCustomJvmArguments) {
                 List<String> cmd = new ArrayList<>();
                 cmd.add("java");
                 cmd.addAll(config.getEffectiveJvmArguments());
                 cmd.addAll(Arrays.asList("-jar", pathToJar));
                 envProducer.produce(KubernetesEnvBuildItem.createSimpleVar("JAVA_APP_JAR", pathToJar, null));
-                commandProducer.produce(KubernetesCommandBuildItem.command(cmd));
+                commandProducer.produce(new KubernetesCommandBuildItem(cmd, config.jvmArguments.get()));
```
Comment (Sgitario) [inline extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java:164]: This is not correct. One thing is the JVM arguments (for example: `-Dprop=key`), and another thing is the program arguments which is addressed with `quarkus.kubernetes.arguments`.

**Review by Sgitario — CHANGES_REQUESTED** [review]
Other necessary changes could be to remove the `OpenshiftConfig.getEffectiveJvmArguments` method which is confusing and does not add much value. 
Also, we have this `getEffectiveJvmArguments` method in JIBConfig and S2IConfig because we had a default value for the jvmArguments, so had two properties `jvmArguments` and `jvmAdditionalArguments`. We should validate that this default value in `jvmArguments` and the `jvmAdditionalArguments` are respected after these changes.
Finally, I'm missing some tests to verify the actual change. As far I understand, before these changes, we only generated a command if we didn't know about the base image. Now, we're always generating the command if a custom jar path or the arguments are set by the user, regardless the base image we're using.
Another last note, this could be a breaking change that should be noted in the release notes.

**Review by Sgitario — APPROVED** [review]
The changes look good to me.
Moreover, I checked the reproducer in the linked issue and now the generated `DeploymentConfig` resource is the same for JVM and Native. So, the app should behave the same.

**Pull request merged**